assets/l10n/app_en.arb

```diff
@@ -27,18 +31,290 @@
   "@profileButtonSendDirectMessage": {
     "description": "Label for button in profile screen to navigate to DMs with the shown user."
   },
-  "cameraAccessDeniedTitle": "Permissions needed",
-  "@cameraAccessDeniedTitle": {
-    "description": "Title for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededError": "Error",
+  "@permissionsNeededError": {
+    "description": "General title for access dialog when we encounter an unknown error."
   },
-  "cameraAccessDeniedMessage": "To upload an image, please grant Zulip additional permissions in Settings.",
-  "@cameraAccessDeniedMessage": {
-    "description": "Message for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededTitle": "Permissions needed",
+  "@permissionsNeededTitle": {
+    "description": "Title for dialog when the user needs to grant additional permissions."
   },
-  "cameraAccessDeniedButtonText": "Open settings",
-  "@cameraAccessDeniedButtonText": {
+  "permissionsNeededOpenSettings": "Open settings",
+  "@permissionsNeededOpenSettings": {
+    "description": "Button label for permissions dialog button that opens the system settings screen."
+  },
+  "permissionsNeededCameraAccessDenied": "To upload an image, please grant Zulip additional permissions in Settings.",
+  "@permissionsNeededCameraAccessDenied": {
     "description": "Message for dialog when the user needs to grant permissions for camera access."
   },
+  "permissionsReadExternalStorageDenied": "To upload files, please grant Zulip additional permissions in Settings.",
+  "@permissionsReadExternalStorageDenied": {
+    "description": "Message for dialog when the user needs to grant permissions for external storage read access."
+  },
+  "actionSheetCopyMessageText": "Copy message text",
+  "@actionSheetCopyMessageText":  {
+    "description": "Label for copy message text button on action sheet."
+  },
+  "actionSheetCouldNotFetchMessageSource": "Could not fetch message source",
+  "@actionSheetCouldNotFetchMessageSource": {
+    "description": "Dialog message when the source of a message could not be fetched."
```

Comment:
For these error messages, I feel like the relevant context isn't so much "action sheet" as that it's the error message to appear in a dialog box.

As a test for making these naming choices, consider if we had some other place in the UI — somewhere other than an action sheet — where you could do something that needed to fetch the source of a message, and that fetching failed. Would we want the message to be different? I think we'd want it to be the same (and therefore save translators' effort re-translating it, and avoid showing users untranslated versions until they do).

So perhaps `errorCouldNotFetchMessageSource`. Potentially could be like `errorDialogCouldNotFetchMessageSource`, but I think the "dialog" specificity wouldn't be helpful either — if we show it in a toast/[snackbar](https://m3.material.io/components/snackbar/overview) or something, we'll still want the same message.